CHANGELOG.md

```diff
@@ -6,6 +6,19 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html),
 with the exception that 0.x versions can break between minor versions.
 
+## [Unreleased]
+### Added
+- Support for relative backreferences using `\k<-1>` (-1 references the
+  previous group) (#121)
+- Add `try_replacen` to `Regex` which returns a `Result` instead of panicking
+  when matching errors (#130)
+### Changed
+- Switch from regex crate to regex-automata and regex-syntax (lower level APIs)
+  to simplify internals (#121)
```

Comment:
Wow this is great! Happy to take feedback if you have it. :-)

Reply:
Hey @BurntSushi :)! Yeah it is! You can see how we could simplify in these places:

* https://github.com/fancy-regex/fancy-regex/pull/121/files#diff-3bc51552cab41d1a2dbf07842cb438088563f6134a9c69a266dfd0d79b631495L177
* https://github.com/fancy-regex/fancy-regex/pull/121/files#diff-3bc51552cab41d1a2dbf07842cb438088563f6134a9c69a266dfd0d79b631495L652

Also using `LookMatcher`:
* https://github.com/fancy-regex/fancy-regex/pull/121/files#diff-3bc51552cab41d1a2dbf07842cb438088563f6134a9c69a266dfd0d79b631495R492

No specific feedback, except it all works really nicely for us, great work :)